jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/component/LifeCycleListenerTest.java

```diff
@@ -222,4 +223,58 @@ public void lifeCycleStopping(LifeCycle event)
             }
         }
     }
+
```

Comment:
The #1 motivation for this new `installBean` method was to make sure we don't de-ref objects under construction. We should find a way to test that.

Reply:
Neither me nor chatGPT can think of a way that the method `addBean` can tell if it has been called from a super constructor or not.   Thus it is impossible for us to test if it is safe for `addBean` to call a listener passing this.     We have provided and alternative to `addBean` that is safe to call from constructors and we have javadoc'd that alternative.

I have visually scanned the list of callers of addBean and can see no constructors in there.  However, I cannot tell if any of those callers are themselves called from a constructor.

I doubt that we are perfect.... but we are better than we were!